src/Domain/CoordinateMaps/FocallyLiftedSide.hpp

```diff
@@ -0,0 +1,309 @@
+ * \f}
+ *
+ * In solving the quadratic, we choose the larger root if
+ * \f$x^2>z_\mathrm{P}\f$ and the smaller root otherwise. We demand
+ * that the root is greater than unity.  If there is no such root,
+ * this means that the point \f$x^i\f$ is not in the range of the map
+ * so we return a default-constructed `std::optional<double>`.
+ *
+ * ### deriv_lambda_tilde
+ *
+ * `deriv_lambda_tilde` takes as arguments \f$x_0^i\f$, a projection point
+ *  \f$P^i\f$, and \f$\tilde{\lambda}\f$, and
+ *  returns \f$\partial \tilde{\lambda}/\partial x^i\f$.
+ * By differentiating Eq. (14), we find
+ *
+ * \f{align}
+ * \frac{\partial\tilde{\lambda}}{\partial x^j} &=
+ * \tilde{\lambda}^2 \frac{C^j - x_0^j}{|x_0^i - P^i|^2
+ * + (x_0^i - P^i)(P_i - C_{i})}.
+ * \f}
+ *
+ * ### inv_jacobian
+ *
+ * `inv_jacobian` returns \f$\partial \bar{x}^i/\partial x_0^k\f$,
+ *  where \f$\sigma\f$ is held fixed.
+ * The arguments to `inv_jacobian` are \f$(\bar{x},\bar{y},\bar{z})\f$.
+ *
+ * Note from Eqs. (9--12) that \f$\bar{x}\f$ and \f$\bar{y}\f$
+ * depend only on \f$x_0^0\f$ and \f$x_0^1\f$ but not on \f$x_0^2\f$.
+ *
+ * By differentiating Eqs. (9--12), we find
+ *
+ * \f{align*}
+ * \frac{\partial \bar{x}}{\partial x_0^0} &=
+ * \frac{\bar{y}^2}{(1+\sigma)\rho},\\
+ * \frac{\partial \bar{x}}{\partial x_0^1} &=
+ * - \frac{\bar{x}\bar{y}}{(1+\sigma)\rho},\\
+ * \frac{\partial \bar{x}}{\partial x_0^2} &= 0,\\
+ * \frac{\partial \bar{y}}{\partial x_0^0} &=
+ * - \frac{\bar{x}\bar{y}}{(1+\sigma)\rho},\\
+ * \frac{\partial \bar{y}}{\partial x_0^1} &=
+ * \frac{\bar{x}^2}{(1+\sigma)\rho},\\
+ * \frac{\partial \bar{y}}{\partial x_0^2} &= 0,\\
+ * \frac{\partial \bar{z}}{\partial x_0^0} &= 0,\\
+ * \frac{\partial \bar{z}}{\partial x_0^1} &= 0,\\
+ * \frac{\partial \bar{z}}{\partial x_0^2} &=
+ * -\frac{2}{\rho(\theta_\mathrm{min}-\theta_\mathrm{max})},
+ * \f}
+ *
+ * where
+ *
+ * \f[
+ *   \rho = R \sin\theta = R\sin\left(\theta_\mathrm{max} +
+ * (\theta_\mathrm{min}-\theta_\mathrm{max}) \frac{\bar{z}+1}{2}\right),
+ * \f]
+ *
+ * which is also equal to the quantity in Eq. (12).
+ *
+ * ### dxbar_dsigma
+ *
+ * `dxbar_dsigma` returns \f$\partial \bar{x}^i/\partial \sigma\f$,
+ *  where \f$x_0^i\f$ is held fixed.
+ *
+ * From Eqs. (10) and (11) we have
+ *
+ * \f{align}
+ * \frac{\partial \bar{x}^i}{\partial \sigma} &=
+ * \left(\frac{\bar{x}}{\sqrt{\bar{x}^2+\bar{y}^2}},
+ *       \frac{\bar{y}}{\sqrt{\bar{x}^2+\bar{y}^2}},0\right).
+ * \f}
+ *
+ */
+class Side {
+ public:
+  static constexpr size_t dim = 3;
+  Side(const std::array<double, 3>& center, double radius, double z_lower,
+       double z_upper) noexcept;
+
+  Side() = default;
+  ~Side() = default;
+  Side(Side&&) = default;
+  Side(const Side&) = default;
+  Side& operator=(const Side&) = default;
+  Side& operator=(Side&&) = default;
+
+  template <typename T>
+  void forward_map(
+      const gsl::not_null<std::array<tt::remove_cvref_wrap_t<T>, 3>*>&
```

Comment:
why a reference to a `not_null`? That's a pointer to a pointer now :P we always just take by value.

below as well